### PR TITLE
GPII-2937: Migration to Electron 1.8.4

### PR DESCRIPTION
**Pull request by danailbd**
In order to use Electron 1.8.4, re-contextualizing for the Electron menu template object is needed.
This is caused by a [change](https://github.com/electron/electron/blob/v1.8.4/lib/browser/api/menu.js#L170) in an Array check to a less graceful one.

Infusion expansion uses a different context which causes this check to fail.

### CODE REVIEW

File: src/main/menu.js [hunk 1]
```diff
@@ -64,6 +64,11 @@ fluid.defaults("gpii.app.menuInApp", {
   * @param events {Object} An object containing the events that may be fired by items in the menu.
   */
 gpii.app.updateMenu = function (tray, menuTemplate, events) {
+    // Needed in order to get around this non graceful check: https://github.com/electron/electron/blob/v1.8.4/lib/browser/api/menu.js#L170
+    // The infusion's expander uses a context different than the current,
```
Comment (javihernandez) [inline src/main/menu.js]: ... uses a different context ...

File: src/main/menu.js [hunk 2]
```diff
@@ -64,6 +64,11 @@ fluid.defaults("gpii.app.menuInApp", {
...
+    // which cases this Array check to fail.
```
Comment (javihernandez) [inline src/main/menu.js]: ... which *causes* ...

File: src/main/utils.js [hunk 3]
```diff
@@ -148,3 +148,21 @@ gpii.app.timer.clear = function (that) {
         that.timer = null;
     }
 };
+
+/**
+ * Set proper context (the current) for elements such as arrays.
+ * This is needed in order for arrays to pass the more strict
+ * check of: `instanceof array`. In generally such checks are to be avoided
```
Comment (javihernandez) [inline src/main/utils.js]: s/generally/general

**Comment by javihernandez** [comment]
Thanks for this update @danailbd :+1: 
Can you update it according to my comments? In general, I'm fine with the code, although the "recontextualise" function looks like a "workaround" to me, but I'll defer to @amb26, maybe you've already talked about this and found out that this is the only way we can achieve such thing at this moment.

**Comment by amb26** [comment]
It's a workaround but it is actually their bug and not ours, and we should file it with them

**Comment by javihernandez** [comment]
> It's a workaround but it is actually their bug and not ours, and we should file it with them

Sure, I wasn't complaining, just double checking.

File: src/main/menu.js (hunk 1 above)
Comment (amb26) [reply src/main/menu.js]: Should be clear that this refers to a V8 JS VM context as documented at https://nodejs.org/api/vm.html

**Comment by javihernandez** [comment]
ok to test

**Comment by javihernandez** [comment]
ok to test

**Comment by javihernandez** [comment]
Tests are passing locally :thinking: 
From the last logs, I'm seeing that the tests are failing right after the "C:\\Windows\\System32\\LicensingUI.exe" process pops up. Also, I'm seeing that CI is using the previous version of our vagrant box (2018.01.25), which has already expired and may be the source of these CI failures.
@amatas, can you nudge our beloved CI to use our last vagrant box?

**Comment by javihernandez** [comment]
ok to test

**Pull request merged**